src/App.jsx

```diff
@@ -1,81 +1,66 @@
-import "./App.css";
-import React, { useEffect } from "react";
-import { useDispatch, useSelector } from "react-redux";
-import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
-import PropTypes from "prop-types";
-import Courses from "./components/Courses/Courses";
-import Header from "./components/Header/Header";
-import Login from "./components/Login/Login";
-import Registration from "./components/Registration/Registration";
-import CreateCourse from "./components/CreateCourse/CreateCourse";
-import CourseInfo from "./components/CourseInfo/CourseInfo";
-import { ENDPOINTS } from "./util/consts";
-import APIService from "./util/APIService";
-import { addCourses } from "./store/courses/actionCreators";
-import { addAuthors } from "./store/authors/actionCreators";
+import './App.css';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+import PrivateRoute from './components/PrivateRouter/PrivateRouter';
+import AdminRoute from './components/AdminRouter/AdminRouter';
+import Courses from './components/Courses/Courses';
+import Header from './components/Header/Header';
+import Login from './components/Login/Login';
+import Registration from './components/Registration/Registration';
+import CourseForm from './components/CourseForm/CourseForm';
+import CourseInfo from './components/CourseInfo/CourseInfo';
+import { getAllAuthors } from './store/authors/thunk';
+import { getAllCourses } from './store/courses/thunk';
 
 function App() {
-  const dispatch = useDispatch();
-  const appstate = useSelector((state) => state);
+	const dispatch = useDispatch();
+	const appstate = useSelector((state) => state);
```

Comment:
It is good practive to create reusable functions (selectors) and put them in useSelector.